PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingDetailVC.swift

```diff
@@ -267,7 +311,26 @@ class PetListingDetailVC: UITableViewController
         return UITableViewCell()
     }
 
-    ////////////////////////////////////////////////////////////
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+
+        let sectionNumber = indexPath.section
+        let rowNumber = indexPath.row
+        
+        if sectionNumber == 3 && rowNumber == 1 {
+            let mailComposeViewController = configureMailController()
+            if MFMailComposeViewController.canSendMail() {
+                self.present((mailComposeViewController)!, animated: true, completion: nil)
```

Comment:
The idea with making the return from `configureMailController()` be optional was that you could use an `if let` and avoid force unwrapping it like you're doing.  Something more like
```swift
if let mailComposeViewController = configureMailController() {
    if MFMailComposeViewController.canSendMail() {
        self.present(mailComposeViewController, animated: true, completion: nil)
    } else {
        displayEmailError()
    }
}
```

Reply:
It's been a while since I looked over at this!  I actually medled with the solution you gave and it works except that all the cells available become active and end up triggering the mailComposeVC.  How should I go about preventing that?

Reply:
Do you mean implementing the above snippet of code?  I just made that change locally (it's not that different than what you currently had there) and it seems to only be trying to compose mail on the mail row.